### ISSUE TITLE
Feature Request: Linux Support

### ISSUE THREAD

**Issue by LumiWasTaken** [Kiln-AI/Kiln#40]
Hiya,

First of all, amazing tool. It's a Cleaner + GUI Version to what i have been privately building. 

i've had a hard time spotting any support for the most commonly used OS in the context of ML / AI Development Linux.

Are there any plans to Release for example an AppImage or guidance on how to build it for Linux?

If not, what are the roadblocks, so assistance could be provided?

Thanks in advance!

**Comment by scosman** [comment]
@LumiWasTaken try the build here: https://github.com/Kiln-AI/Kiln/actions/runs/11947069712

Linux should work, I just don't have an environment setup for testing. Note: pyinstaller builds raw executables, not AppImage or FlatPak. Might be too much of a pain for typical users?

**Comment by LumiWasTaken** [comment]
Hi, that worked pretty good! I suppose a nice to have would be automatically opening the default Browser with the localhost:8757 port.

Other than that this worked perfectly and i enjoy it.

I was tossing in the term "AppImage" given that i'm not 100% confident what would be the best solution.

Though i have a few more "Improvements" that i'll bring up in the different issues to make this single-topic.

TL;DR
It started, it worked. Thank you!

**Comment by scosman** [comment]
Not done yet, still in a branch!

@LumiWasTaken mind trying this one and reporting back? It uses a single file distribution which simplifies things for users moving things around: https://github.com/Kiln-AI/Kiln/actions/runs/11953608049

Other questions:
 - does the tray icon work and look okay?
 - does the app icon work and look okay?

An appimage would be ideal if we want to do this properly, but single file pack can work for now.

**Comment by LumiWasTaken** [comment]
@scosman Oops!

Sorry! In a bit of a rush today.

When i hopen the Klin binairy i get the following:
```
./Kiln 
The environment does not allow connecting to the splash screen. Did bootloader fail to initialize it?
Traceback (most recent call last):
  File "PyInstaller/fake-modules/pyi_splash.py", line 108, in <module>
  File "<frozen os>", line 714, in __getitem__
KeyError: '_PYI_SPLASH_IPC'
```

First I thought that's just a normal issue but now that you've mentioned Try Icon etc. I do have an Empty spot in my try and no icon etc. available.

When using CTRL + C i have the following output before it closes

```
^C^C^C^CTraceback (most recent call last):
  File "desktop/desktop.py", line 108, in <module>
  File "tkinter/__init__.py", line 1505, in mainloop
KeyboardInterrupt
[PYI-79057:ERROR] Failed to execute script 'desktop' due to unhandled exception!
^CException ignored in: <object repr() failed>
Traceback (most recent call last):
  File "threading.py", line 1624, in _shutdown
KeyboardInterrupt: 
```

I am currently on X11 and not Wayland using the KDE DE on EndavourOS

**Comment by LumiWasTaken** [comment]
Note: I was only testing functionality before, I did not realize that the tray icon etc. were relevant.

**Comment by scosman** [comment]
There is always something to tweak on a new platform 😀 

Any interest in contributing?

Things to handle I think are: 

 - Get the tray icon working (or disable it). The package we use [pystray](https://pystray.readthedocs.io/en/latest/usage.html#supported-backends) supports Linux, so it should work. Just needs debugging. See that 
 - Get the app icon working. 
 - Splash screen log is strange, we only enable the splash screen on Windows. P3 if it isn't impacting functionality
 - P2: AppImage
 - Check the Kiln Projects folder is creating in the user directory correctly
 - The KeyboardInterrupt is find I think. That kinda makes sense. If the tray works folks can gracefully exit.

**Comment by scosman** [comment]
Partially done: https://github.com/Kiln-AI/Kiln/pull/41

I think the "real" fix is to use Tauri with python sidecar. Too many hacks to get taskbar and packaging working properly. 

Still, it works pretty well for no customization for the platform!

**Comment by LumiWasTaken** [comment]
Hi,
thank you so much, it works pretty neat. I think the taskbar issue is irrelevant to most linux users as they would probably run it inside their Terminal anyway and just Ctrl + C out of it to kill it.

From my Point of view the addition of such Taskbar icon is unnessecairy and just like Ollama did for Linux, focusing on the terminal is more than fine and maybe even an AppImage would be overkill.

**Comment by scosman** [comment]
Closing since we have experimental support out.
